Add CODEOWNERS usage details (#7784)
github/docs: content/github/creating-cloning-and-archiving-repositories/creating-a-repository-on-github/about-code-owners.md

# directory in the root of your repository and any of its
# subdirectories.
/docs/ @doctocat

# In this example, @octocat owns any file in the `/apps` 
# directory in the root of your repository except for the `/apps/github` 

> # repository, as its owners are left empty.

Reviewer: Spotted a typo

```suggestion
# subdirectory, as its owners are left empty.
```

# directory in the root of your repository and any of its
# subdirectories.
/docs/ @doctocat

# In this example, @octocat owns any file in the `/apps` 
# directory in the root of your repository except for the `/apps/github` 
# repository, as its owners are left empty.
/apps/ @octocat
/apps/github 
```
### Syntax exceptions
There are some syntax rules for gitignore files that do not work in CODEOWNERS files:
- Escaping a pattern starting with `#` using `\` so it is treated as a pattern and not a comment
- Using `!` to negate a pattern
- Using `[ ]` to define a character range

## CODEOWNERS and branch protection

> It is possible to add a branch protection rule to ensure that changed code is reviewed by the owners of the changed files, see "[About protected branches](/administering-a-repository/defining-the-mergeability-of-pull-requests/about-protected-branches)." 

Reviewer: Some minor updates for consistency with our style guide:

```suggestion
Repository owners can add branch protection rules to ensure that changed code is reviewed by the owners of the changed files. For more information, see "[About protected branches](/github/administering-a-repository/defining-the-mergeability-of-pull-requests/about-protected-branches)." 
```

### Example of a CODEOWNERS file
```

> # In this example, any change inside `/apps` directory
> # will require approval from @doctocat.
> /apps/ @doctocat
> 
> # In this example, any change inside `/apps` directory
> # will require approval from @doctocat or @octocat.
> /apps/ @doctocat @octocat

Reviewer: ```suggestion
# In this example, any change inside the `/apps` directory
# will require approval from @doctocat.
/apps/ @doctocat

# In this example, any change inside the `/apps` directory
# will require approval from @doctocat or @octocat.
/apps/ @doctocat @octocat
```

> # In this example, assuming @doctocat is a member 
> # of @Github/content-team and @doctocat introduces a 
> # change inside `/apps` directory, the approval 
> # can be done by anyone else, including the users 
> # that are not in @Github/content-team. 
> # @Github/content-team will be still appointed
> # to the review, but it's member's approval
> # is not required. 
> /apps/ @Github/content-team

Reviewer: I've re-written this a bit to try and simplify it. 

```suggestion
# In this example, any change inside the `/apps` directory
# will require approval from a member of the @example-org/content team.
# If a member of @example-org/content opens a pull request 
# with a change inside the `/apps` directory, their approval is implicit.
# The team is still added as a reviewer but not a required reviewer.
# Anyone can approve the changes.
/apps/ @example-org/content-team
```
